lib/fake_braintree/sinatra_app.rb

```diff
@@ -170,16 +171,11 @@ def hash_from_request_body_with_key(key)
       if FakeBraintree.decline_all_cards?
         gzipped_response(422, FakeBraintree.create_failure.to_xml(root: 'api_error_response'))
       else
-        transaction = hash_from_request_body_with_key('transaction')
+        data = hash_from_request_body_with_key('transaction')
         transaction_id = md5("#{params[:merchant_id]}#{Time.now.to_f}")
-        options = transaction["options"] || {}
-        status = "authorized"
-        if options.fetch("submit_for_settlement", false) == true
-          status = "submitted_for_settlement"
-        end
-        transaction_response = {'id' => transaction_id, 'amount' => transaction['amount'], 'status' => status, 'type' => 'sale'}
-        FakeBraintree.registry.transactions[transaction_id] = transaction_response
-        gzipped_response(200, transaction_response.to_xml(root: 'transaction'))
+        transaction = FakeBraintree::Transaction.new(data, transaction_id)
+        response = transaction.create
+        gzipped_response(200, response.to_xml(root: 'transaction'))
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.